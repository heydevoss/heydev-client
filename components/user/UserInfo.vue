<template>
  <div class="content">
    <avatar :avatar-url="avatarUrl" class="user--avatar" />
    <div class="user--info">
      <span class="user--name">{{ name }}</span>
      <span>{{ bio }}</span>
      <span class="user--location">
        <i class="el-icon-location" /> {{ location }}
      </span>
    </div>
  </div>
</template>

<script>
import Avatar from '@/components/commons/Avatar'

export default {
  name: 'UserInfo',
  components: {
    Avatar
  },
  props: {
    name: { type: String, default: '' },
    bio: { type: String, default: '' },
    location: { type: String, default: '' },
    avatarUrl: { type: String, default: '' }
  }
}
</script>

<style lang="scss" scoped>
.content {
  display: flex;
  align-items: center;
}
.user--info {
  margin-left: 40px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}
.user--name {
  font-size: 2rem;
  margin-bottom: 5%;
}
.user--location {
  margin-top: 40px;
}
.user--avatar {
  width: 150px;
  height: 150px;
}

@media screen and (max-width: 820px) {
  .user--avatar {
    width: 60px;
    height: 60px;
  }
}
</style>
