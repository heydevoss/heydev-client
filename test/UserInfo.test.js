import { shallowMount } from '@vue/test-utils'
import UserInfo from '@/components/user/UserInfo'

describe('<UserInfo/> Component', () => {
  const wrapper = shallowMount(UserInfo)

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
