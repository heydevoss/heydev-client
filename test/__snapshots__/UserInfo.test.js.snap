// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInfo/> Component renders correctly 1`] = `
<div
  class="content"
>
  <avatar-stub
    class="user--avatar"
  />
   
  <div
    class="user--info"
  >
    <span
      class="user--name"
    >
      
    </span>
     
    <span>
      
    </span>
     
    <span
      class="user--location"
    >
      <i
        class="el-icon-location"
      />
       
    
    </span>
  </div>
</div>
`;
